arangod/Aql/IResearchViewExecutor.cpp

```diff
@@ -1045,6 +1084,39 @@ size_t IResearchViewExecutor<ordered, materializeType>::skip(size_t limit) {
     _doc = nullptr;
   }
 
+  saveCollection();
+
+  return toSkip - limit;
+}
+
+template <bool ordered, MaterializeType materializeType>
+size_t IResearchViewExecutor<ordered, materializeType>::skipAll() {
+  TRI_ASSERT(this->_indexReadBuffer.empty());
+  TRI_ASSERT(this->_filter);
+
+  size_t skipped = 0;
+
+  for (size_t count = this->_reader->size(); _readerOffset < count;) {
+    if (!_itr && !resetIterator()) {
+      continue;
+    }
+
+    while (_itr->next()) {
+      skipped++;
+    }
+
+    ++_readerOffset;
+    _itr.reset();
+    _doc = nullptr;
+  }
+
+  saveCollection();
```

Comment:
If I got things correctly, after skipAll no produceRows possible, so saving collection there might be redundant

Reply:
Thanks. Done